compiler/aarch64/codegen/BinaryEvaluator.cpp

```diff
@@ -233,19 +234,52 @@ OMR::ARM64::TreeEvaluator::lushrEvaluator(TR::Node *node, TR::CodeGenerator *cg)
 	return OMR::ARM64::TreeEvaluator::unImpOpEvaluator(node, cg);
 	}
 
+// also handles lrolEvaluator
 TR::Register *
 OMR::ARM64::TreeEvaluator::irolEvaluator(TR::Node *node, TR::CodeGenerator *cg)
-	{
-	// TODO:ARM64: Enable TR::TreeEvaluator::irolEvaluator in compiler/aarch64/codegen/TreeEvaluatorTable.hpp when Implemented.
-	return OMR::ARM64::TreeEvaluator::unImpOpEvaluator(node, cg);
-	}
+   {
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Node *secondChild = node->getSecondChild();
+   TR::Register *srcReg = cg->evaluate(firstChild);
+   TR::Register *trgReg = cg->allocateRegister();
```

Comment:
This should be worthy of a check to see if the `srcReg` dies here, and if so simply re-use it for `trgReg`.